src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -906,6 +940,32 @@ SDCPUThread::onIssueAccessed(weak_ptr<InflightInst> inst)
     inst_ptr->traceData(nullptr);
 #endif
 
+    if (inst_ptr->staticInst()->isLastMicroop()) {
+        DPRINTF(SDCPUThreadEvent, "Releasing MacroOp after decoding final "
+                                  "microop\n");
+        curMacroOp = StaticInst::nullStaticInstPtr;
+    }
+
+    const bool interrupt_detected = _cpuPtr->checkInterrupts(this);
+    bool need_advance = true;
+    if (interrupt_detected) {
+        DPRINTF(SDCPUThreadEvent, "Interrupt detected at issue time, clearing "
+                                  "buffer as much as possible.\n");
+        need_advance = false;
+        for (auto& other_inst_ptr : inflightInsts) {
```

Comment:
What happens if the interrupt is handled when there are some instructions on the inflight_inst buffer. I'm worried that it won't be 'consistent' since gem5's interrupt handlers will assume that there are no active instructions in flight (I think this is true). I think we should start with just checking at commit time.

Reply:
The interrupt isn't handled at that time, it's checked during issue, just to stop the front-end from continuing beyond what is necessary if an interrupt is detected. The actual handling of the interrupt is delayed until commit time, specifically at the first commit of an operation that is not marked isDelayedCommit().